Source/WebCore/html/ImageBitmap.cpp

```diff
@@ -194,6 +194,11 @@ void ImageBitmap::prepareForCrossThreadTransfer()
 {
     m_bitmap = ImageBuffer::sinkIntoImageBufferForCrossThreadTransfer(WTFMove(m_bitmap));
 }
+
+void ImageBitmap::finalizeCrossThreadTransfer()
+{
+    m_bitmap = ImageBuffer::sinkIntoImageBufferAfterCrossThreadTransfer(WTFMove(m_bitmap));
```

Comment:
Can we save the dance if there was no cross thread transfer?

Reply:
technically yes, but I think the original idea of all that `sink*` functions was to encapsulate all the logic in `ImageBuffer`